packages/grafana-ui/src/components/RadialGauge/colors.ts

```diff
@@ -0,0 +1,179 @@
+    ];
+  }
+
+  const colorMode = getFieldColorMode(fieldDisplay.field.color?.mode);
+
+  // thresholds get special handling
+  if (colorMode.id === FieldColorModeId.Thresholds) {
+    const displayProcessor = getFieldDisplayProcessor(fieldDisplay);
+    const [min, max] = getFieldConfigMinMax(fieldDisplay);
+    const thresholds = fieldDisplay.field.thresholds?.steps ?? [];
+
+    const result: Array<{ color: string; percent: number }> = [
+      { color: displayProcessor(min).color ?? baseColor, percent: 0 },
+    ];
+
+    for (const threshold of thresholds) {
+      if (threshold.value > min && threshold.value < max) {
+        const percent = (threshold.value - min) / (max - min);
+        result.push({ color: theme.visualization.getColorByName(threshold.color), percent });
+      }
+    }
+
+    result.push({ color: displayProcessor(max).color ?? baseColor, percent: 1 });
+
+    return result;
+  }
+
+  // Handle continuous color modes before other by-value modes
+  if (colorMode.isContinuous && colorMode.getColors) {
+    const colors = colorMode.getColors(theme);
+    return colors.map((color, idx) => ({ color, percent: idx / (colors.length - 1) }));
+  }
+
+  // For value-based colors, we want to stay more true to the specific color,
+  // so a radial gradient that adds a bit of light and shade works best
+  if (colorMode.isByValue) {
+    const darkerColor = tinycolor(baseColor).darken(5);
+    const lighterColor = tinycolor(baseColor).spin(20).lighten(10);
+
+    const color1 = theme.isDark ? lighterColor : darkerColor;
+    const color2 = theme.isDark ? darkerColor : lighterColor;
+
+    return [
+      { color: color1.toString(), percent: 0 },
+      { color: color2.toString(), percent: 0.6 },
+      { color: color2.toString(), percent: 1 },
+    ];
+  }
+
+  // For fixed / palette based color scales we can create a more hue and light
+  // based linear gradient that we rotate with the value
+  const darkerColor = tinycolor(baseColor)
+    .spin(-20)
+    .darken(theme.isDark ? 15 : 5);
+  const lighterColor = tinycolor(baseColor).saturate(20).spin(20).brighten(10).lighten(10);
+
+  const underlyingGradient = [
+    { color: theme.isDark ? darkerColor.toString() : lighterColor.toString(), percent: 0 },
+    { color: theme.isDark ? lighterColor.toString() : darkerColor.toString(), percent: 1 },
+  ];
+
+  // rotate the gradient so that the highest contrasting point is the value, depending on theme.
+  const valuePercent = getValuePercentageForValue(fieldDisplay);
+  const startColor = theme.isDark
+    ? colorAtGradientPercent(underlyingGradient, 1 - valuePercent).toHexString()
+    : underlyingGradient[0].color;
+  const endColor = theme.isDark
+    ? underlyingGradient[1].color
+    : colorAtGradientPercent(underlyingGradient, valuePercent).toHexString();
+  return [
+    { color: startColor, percent: 0 },
+    { color: endColor, percent: valuePercent },
+    { color: endColor, percent: 1 },
+  ];
```

Comment:
regarding the fixed color gradients, because the "point" mode for endpoint markers needs to be able to tell exactly what the color is going to be, we can't use CSS to rotate the gradient, we need to calculate it like this so that that calculation can do its thing.